src/sinks/file.rs

```diff
@@ -0,0 +1,316 @@
+use crate::{
+    buffers::Acker,
+    event::Event,
+    sinks::util::{
+        encoding::BasicEncoding,
+        file::{EmbeddedFileSink, FileSink},
+        SinkExt,
+    },
+    template::Template,
+    topology::config::DataType,
+};
+
+use futures::{future, Async, AsyncSink, Sink, StartSend};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::path::PathBuf;
+use std::time::Instant;
+
+#[derive(Deserialize, Serialize, Debug)]
+#[serde(deny_unknown_fields)]
+pub struct FileSinkConfig {
+    #[serde(rename = "path")]
+    pub path_template: String,
+    #[serde(rename = "timeout", default = "default_close_timeout_secs")]
+    pub close_timeout_secs: Option<u64>,
+    pub encoding: Option<BasicEncoding>,
+}
+
+fn default_close_timeout_secs() -> Option<u64> {
+    None //means that we don't close files, so no overhead
+}
+
+impl FileSinkConfig {
+    pub fn new(path_template: String) -> Self {
+        Self {
+            path_template,
+            close_timeout_secs: default_close_timeout_secs(),
+            encoding: None,
+        }
+    }
+}
+
+#[typetag::serde(name = "file")]
+impl crate::topology::config::SinkConfig for FileSinkConfig {
+    fn build(&self, acker: Acker) -> Result<(super::RouterSink, super::Healthcheck), String> {
+        let sink = PartitionedFileSink::new(
+            Template::from(&self.path_template[..]),
+            self.close_timeout_secs,
+            self.encoding.clone(),
+        )
+        .stream_ack(acker);
+
+        Ok((Box::new(sink), Box::new(future::ok(()))))
+    }
+
+    fn input_type(&self) -> DataType {
+        DataType::Log
+    }
+}
+
+pub struct PartitionedFileSink {
+    path_template: Template,
+    encoding: Option<BasicEncoding>,
```

Comment:
So this option should be non optional since there is _always_ an encoding type and they should be unwrapped here https://github.com/timberio/vector/pull/688/files#diff-daf9b4d6c1509d82cde192a4dfcffc73R45

Reply:
Forcing sinks to always have a value makes sense, but I see that in most of other sinks `Option` is used and sometimes `None` is treated as Text. Are they outdated about this point?

Reply:
Good point, we should definitely be consistent about this and open any issues to fix where we are not.